### DISCUSSION TITLE
Getting an error with sidebar icons during the build.

### DISCUSSION THREAD

**Discussion by 3p5ilon** [fuma-nama/fumadocs#1130] [Q&A] [answered]
I liked the design of the Fumadocs sidebar icons, so I copied the Fumadocs code, but it's throwing an error in production.

```tsx 
// icon.tsx
import type { LucideIcon } from 'lucide-react';
import { TerminalIcon } from 'lucide-react';
import { type HTMLAttributes } from 'react';
import { cn } from '@/utils/cn';

export default function IconContainer({
  icon: Icon,
  ...props
}: HTMLAttributes<HTMLDivElement> & {
  icon?: LucideIcon;
}): React.ReactElement {
  return (
    <div
      {...props}
      className={cn(
        'rounded-md border bg-gradient-to-b from-muted to-secondary p-0.5 shadow-md [a[data-active=true]_&]:from-primary/60 [a[data-active=true]_&]:to-primary [a[data-active=true]_&]:text-primary-foreground',
        props.className,
      )}
    >
      {Icon ? <Icon /> : <TerminalIcon />}
    </div>
  );
}
```

```ts
// lib/source.ts
import { docs, meta } from "@/.source";
import { createMDXSource } from "fumadocs-mdx";
import { loader } from "fumadocs-core/source";
import { icons } from "lucide-react";
import { createElement } from "react";
import IconContainer from "@/app/components/icon";

export const source = loader({
  baseUrl: "/docs",
  source: createMDXSource(docs, meta),
  icon(icon) {
    if (icon && icon in icons)
      return createElement(IconContainer, {
        icon: icons[icon as keyof typeof icons],
      });
  },
});
```

```ts
// utils/cn.ts
export { twMerge as cn } from 'tailwind-merge';
```

Error on build: 
<img width="1205" alt="Screenshot 2024-12-01 at 1 48 50 AM" src="https://github.com/user-attachments/assets/f0f17530-8e72-49ef-ae0b-b1adbbabee81">

**Comment by 3p5ilon** [comment]
GitHub: https://github.com/1msirius/odlai

**Comment by fuma-nama** [comment, marked as answer]
You putted the components folder in app folder, Next.js treated it as a route handler

**Reply by 3p5ilon** [reply]
Oh, my bad! It's working fine now. 

But why isn't the theme sidebar Docs Change section icon theme working? Am I missing something? The code is still the same as before.

<img width="260" alt="Screenshot 2024-12-01 at 7 02 34 AM" src="https://github.com/user-attachments/assets/f2428317-7c3b-4eba-9682-674d5f778d3a">
<img width="260" alt="Screenshot 2024-12-01 at 7 02 28 AM" src="https://github.com/user-attachments/assets/a899e97d-d0c3-43c4-97b5-3ac0814cd32b">

**Comment by fuma-nama** [comment]
we add decorations to the icons via `sidebar.tabs.transform` option in docs layout

**Reply by 3p5ilon** [reply]
It’s working fine, but the color theme isn’t changing between dark and light modes since I initialized the color. 

<img width="260" alt="Screenshot 2024-12-01 at 9 55 59 AM" src="https://github.com/user-attachments/assets/9ec7795b-a7a0-4ab0-87a0-4b80a0d140fc">
<img width="260" alt="Screenshot 2024-12-01 at 9 56 03 AM" src="https://github.com/user-attachments/assets/f5c37391-01c5-4187-b400-7e7c2435b497">

```tsx
// my code
 icon: (
            <Slot
              className="bg-gradient-to-t from-fd-background/80 p-1 [&_svg]:size-5"
              style={{
                color: `#89B4F9`,
                backgroundColor: `rgba(137, 180, 249, 0.2)`,
              }}
            >
              {node.icon}
            </Slot>
          )
```

Fumadocs uses meta.file.dirname to change the theme so what is the color code for blue color?

**Reply by fuma-nama** [reply]
Why don't you setup your own color with tailwind?
